application/src/main/java/uk/nhs/hee/tis/revalidation/mapper/DoctorsForDbMapper.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package uk.nhs.hee.tis.revalidation.mapper;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import uk.nhs.hee.tis.revalidation.dto.TraineeInfoDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+
+@Mapper(componentModel = "spring")
+public interface DoctorsForDbMapper {
+
+  @Mapping(source = "designatedBodyCode", target = "connectionStatus", qualifiedByName = "desgnatedBodyToConnectionStatus")
+  @Mapping(source = "designatedBodyCode", target = "designatedBody")
+  TraineeInfoDto toTraineeInfoDto(DoctorsForDB doctor);
+
+  @Named("desgnatedBodyToConnectionStatus")
+  default String designatedBodyToConnectionStatus(String designatedBody) {
+    return (designatedBody == null || designatedBody.equals("")) ? "No" : "Yes";
```

Comment:
`designatedBody == null || designatedBody.equals("")` can be replaced by `StringUtils.isEmpty(designatedBody)`

Reply:
I was tempted to use `isBlank()` but decided that I wouldn't modify behaviour.  I agree that `isEmpty()` is preferable to what is there now so will update.  The SpringUtils version used across the project is deprecated in favour of `hasLength()`.  Is that preferable to using the Apache Commons `StringUtils.isEmpty()`.

Reply:
Yeah, I think Apache Commons is okay.